### PR TITLE
Weights for YoloX Yolo v6 and Yolo v8

### PR DESCRIPTION
**Pull request by Abdurrahheem**
This PR extends download functionality for weights of yolox model from google drive

### Problem
Yolox model weights are required for test in opencv repository (PR with the test is yet to be opened)

### CODE REVIEW

File: testdata/dnn/download_models.py [hunk 1]
```diff
@@ -936,6 +936,11 @@ def handle_bad_download(self):
         gdrive='1rm3mIqjJNu0xPTCjMKnXccspazV1B2zv',
         sha='fcd0fa401c83bf2b29e18239a9c2c989c9b8669d',
         filename='onnx/models/yolov7_not_simplified.onnx'),
+    Model(
+        name='yolox_s_inf_decoder',
+        url='https://drive.google.com/u/0/uc?id=12dVy3ob7T4fYHOkLYnrpUmlysq6JEc5P&export=download',
```
Comment (fengyuentau) [inline testdata/dnn/download_models.py:941]: Is it from our shared drive?

Comment (Abdurrahheem) [reply testdata/dnn/download_models.py:941]: Yes

**Comment by fengyuentau** [comment]
So are you going to add a test case that uses this model? Currently it is added but not used.

**Comment by Abdurrahheem** [comment]
> So are you going to add a test case that uses this model? Currently it is added but not used.

You are right. Test is not yet added. Waiting for this [PR](https://github.com/opencv/opencv/pull/24569) to finish to add a PR with test

**Comment by fengyuentau** [comment]
Ok, lets hold until you submit a new PR.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by mshabunin** [comment]
Maybe it would be easier to add two new models (6 and 8) to #1129 to avoid merge conflicts?

**Comment by Abdurrahheem** [comment]
@asmorkalov @mshabunin I propose to close this PR as the same addiction are in #1129 are present. No need for duplication?

**Comment by asmorkalov** [comment]
Replaced by https://github.com/opencv/opencv_extra/pull/1129

**Pull request closed without merging**